source/core/Events.cpp

```diff
@@ -123,6 +127,16 @@ class EventQueueObject {
     void SetStatus(QIDStatus status) { _status = status; }
 
  private:
+    bool DiscardTopEvent() {
+        if (_eventQueue.size() > 0) {
+            EventData &eventData = _eventQueue.front();
+            eventData.eventDataType->ClearData(eventData.pEventData);
+            THREAD_TADM()->Free(eventData.pEventData);
```

Comment:
We should probably have a single cleanup method in EventData that does both eventDataType->ClearData and Free(eventData.pEventData)

Reply:
Done.